src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1500,4 +1504,53 @@ class TcgStatics {
     }
   }
 
+  /**
+   * Scoop up PokemonCardSets if not blocked by Scoop-Up Block
+   *
+   * @param params Optional settings map
+   *  + pokemonOnly: boolean - if true, scoops up all cards, else all pokemon cards. discards the rest.
+   *  + only: CardList - only scoops up them, discards the rest.
+   * @param target PokemonCardSet to work on
+   * @param delegate Effect delegate used to determine most sources automatically, and to get the card name for the Scoop-Up Block message
+   * @param source Allows you to specify the source of the scoop up. Use intended manually setting SRC_ABILITY.
+   */
+  static void scoopUpPokemon(params=[:], PokemonCardSet target, Object delegate, Source source=null) {
+    if (source == null) {
+      if (delegate.thisObject.cardTypes.is(TRAINER)) source = TRAINER_CARD
+      if (delegate.thisObject.cardTypes.is(POKEMON)) source = ATTACK
+      if (delegate.thisObject.cardTypes.is(ENERGY)) source = SRC_SPENERGY
+    }
+    if (bg.em().retrieveObject("ScoopUpBlock_Count$target.owner.opposite") && target.numberOfDamageCounters && !hasThetaStop(target)) {
+      bc "Scoop-Up Block prevents $delegate.thisObject.name's effect."
+      return
+    }
+    targeted(target, source) {
+      CardList pokemonList = []
+      CardList otherList = []
+      otherList.addAll(target.cards.filterByType(TRAINER, ENERGY))
+      pokemonList.addAll(target.cards.filterByType(POKEMON))
+      if (params.pokemonOnly || params.only) {
+        if (params.only) {
+          if (params.only instanceof Card) params.only = new CardList(params.only)
+          bc "Scooped up ${params.only.getExcludedList(otherList)}"
+        }
+        else {
+          params.only = new CardList(pokemonList)
+          bc "Scooped up $target"
+        }
+        def toDiscard
+        toDiscard = pokemonList.getExcludedList(params.only as CardList)
+        target.owner.pbg.discard.addAll(toDiscard)
+        target.owner.pbg.hand.addAll(pokemonList.getExcludedList(toDiscard as CardList))
+        toDiscard = otherList.getExcludedList(params.only as CardList).discard()
+        otherList.getExcludedList(toDiscard).moveTo(target.owner.pbg.hand)
+      }
+      else {
+        target.owner.pbg.hand.addAll(pokemonList)
+        otherList.moveTo(target.owner.pbg.hand)
+      }
```

Comment:
I wasn't too happy with the readability of the internal logic. Do you think this would work? 

```suggestion

CardList toHand
if(params.only) {
  if (params.only instanceof Card) toHand = new CardList(params.only)
  else if (params.only instanceof CardList) toHand = params.only as CardList
  else throw new IllegalArgumentException("scoopUpPokemon() params.only=${params.only} type not supported")
} else if(params.pokemonOnly) {
  toHand = target.cards.filterByType(POKEMON)
} else {
  toHand = new CardList(target.cards)
}
CardList toDiscard = target.cards.getExcludedList(toHand)

bc "Scooped up ${toHand}"

CardList toHand2 = toHand.filterByType(POKEMON)
target.owner.pbg.hand.addAll(toHand2)
toHand.getExcludedList(toHand2).moveTo(target.owner.pbg.hand)

CardList toDiscard2 = toDiscard.filterByType(POKEMON)
target.owner.pbg.discard.addAll(toDiscard2)
toDiscard.getExcludedList(toDiscard2).discard()
```

Reply:
This seems to work fine. I want to test a couple more things with it, but will likely commit that suggestion.